datastore/storagebackend/postgresql/postgresql.go

```diff
@@ -120,3 +152,50 @@ func createSetFilter(colName string, vals []string) string {
 	}
 	return fmt.Sprintf("(%s IN (%s))", colName, strings.Join(vals, ","))
 }
+
+// cleanup performs various cleanup tasks, like removing old observations from the database.
+func cleanup(db *sql.DB) error {
+
+	// start transaction
+	tx, err := db.Begin()
+	if err != nil {
+		return fmt.Errorf("db.Begin() failed: %v", err)
+	}
+	defer tx.Rollback()
+
+	// remove observations outside valid range
+	loTime, hiTime := common.GetValidTimeRange()
+	cmd := fmt.Sprintf(`
+		DELETE FROM observation
+		WHERE (obstime_instant < to_timestamp(%d))
+		   OR (obstime_instant > to_timestamp(%d))
+	`, loTime.Unix(), hiTime.Unix())
+	_, err = tx.Exec(cmd)
+	if err != nil {
+		return fmt.Errorf("tx.Exec() failed: %v", err)
+	}
+
+	// DELETE FROM time_series WHERE <no FK refs from observation anymore> ... TODO!
+	// DELETE FROM geo_points WHERE <no FK refs from observation anymore> ... TODO!
+
+	// commit transaction
+	if err = tx.Commit(); err != nil {
```

Comment:
I don't know enough about postgres transactions..., if this locks the table, it might be problematic.
But maybe we should not worry about this now.

Reply:
Agreed. I was a bit unsure whether a transaction is really necessary here at all. If we drop the transaction, I guess we need to figure out if there are any potential scenarios that would result in critical race conditions.